src/ExpressionEvaluator/Core/Source/ExpressionCompiler/InstructionDecoder.cs

```diff
@@ -7,13 +8,11 @@
 
 namespace Microsoft.CodeAnalysis.ExpressionEvaluator
 {
-    internal abstract class InstructionDecoder
-    {
-        internal abstract string GetName(DkmClrInstructionAddress instructionAddress, bool includeParameterTypes, bool includeParameterNames, ArrayBuilder<string> argumentValues);
-        internal abstract string GetReturnType(DkmClrInstructionAddress instructionAddress);
-    }
-
-    internal abstract class InstructionDecoder<TMethodSymbol> : InstructionDecoder where TMethodSymbol : class, IMethodSymbol
+    internal abstract class InstructionDecoder<TMethodSymbol, TModuleSymbol, TTypeSymbol, TTypeParameterSymbol>
+        where TMethodSymbol : class, IMethodSymbol
```

Comment:
Consider removing the ISymbol constraints so that implementations do use the interface methods.